### PR TITLE
Add dasherized-test-name-generator-option RFC

### PR DESCRIPTION
**Pull request by GabrielCW**
See emberjs/ember.js#14915

### CODE REVIEW

File: 0000-dasherized-test-name-generator-option.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+(1) Add an option to the test generating commands to allow for the test name to be dasherized instead of being humanized (as it is [at the moment](https://github.com/ember-cli/ember-cli-test-info/blob/master/index.js#L27)), for instance `--dasherize`
+
+OR
+
+(2) Dasherize the test name by default and provide an option to humanize it, to get today's output, for instance `--humanize`
+
+Test name examples for `some-multi-word-service-test.js` :
+
+* With `--dasherize`: `Unit | Service | some-multi-word-service`
+* With `--humanize` (default today): `Unit | Service | some multi word service`
```
Comment (Turbo87) [inline 0000-dasherized-test-name-generator-option.md]: IMHO this RFC should just set the new default to `dasherize`, but without adding any new options that would increase the complexity

Comment (GabrielCW) [inline 0000-dasherized-test-name-generator-option.md]: Best case scenario for me ;)

**Comment by cibernox** [comment]
I also agree that humanized names make no sense. I'm all in to change it.

**Comment by rwjblue** [comment]
Heh, I actually think we should remove the description completely...

**Comment by rwjblue** [comment]
But if we can't do that, we should use dasherized version.

File: 0000-dasherized-test-name-generator-option.md (hunk 1 above)
Comment (rwjblue) [inline 0000-dasherized-test-name-generator-option.md]: Agreed, this RFC should only be about changing the defaults.  I am generally 👎 on adding new options...

**Comment by GabrielCW** [comment]
According to your feedback, I updated the RFC by removing the options.

The fix is also simpler, although I don't know the uses of `ember-cli-test-info` well enough to know if the `humanize()` function is called elsewhere, as it is public.

**Comment by Turbo87** [comment]
> Heh, I actually think we should remove the description completely... But if we can't do that, we should use dasherized version.

@rwjblue if we pull in https://github.com/emberjs/ember-qunit/pull/258 then removing the test description is not an option

can we do anything to move this forward? I'm constantly being trolled by this when trying to test only single components...

**Comment by rwjblue** [comment]
> @rwjblue if we pull in emberjs/ember-qunit#258 then removing the test description is not an option

I think you misunderstand me, I am saying we do not need **both** a _name_ **and** a _description_.  We should only have one, and actually it matches better with https://github.com/emberjs/ember-qunit/pull/258!!

**Comment by Turbo87** [comment]
@rwjblue indeed, I still don't understand what you mean. what are your definitions of "name" and "description"?

**Comment by rwjblue** [comment]
@GabrielCW - Thank you very much for taking the time to write up this RFC!  

I think it's very clearly correct, and in our opinion points out some pretty bad developer ergonomics in the current default generation system. It has been the policy for quite some time in Ember-land to consider bad developer ergonomics to be bugs.  

I am going to close this RFC as unneeded and advocate for landing the changes proposed in this RFC as a straight up bug fix.

Thank you again for helping push this forward!

**Pull request closed without merging**